### DISCUSSION TITLE
Помогите с ввбором роутера

### DISCUSSION THREAD

**Discussion by Mihrutkin** [bol-van/zapret#228] [Q&A] [answered]
Сейчас tp-link ax23, в нем всего 16mb disk и этого не хватает для установки bol-van.
Помогите выбрать роутер, подходящий для bol-van?

**Comment by fedya** [comment]
если усб порт есть то достаточно добавить флешку отформатированную в ext2

**Comment by Mihrutkin** [comment]
Как раз на моем роутере флеш нет.

**Comment by bol-van** [comment]
прога называется запрет а не болван
болван это автор

16 мб флэша должно хватать в опенврт
разбирайтесь чем забита память

если вы ставите гит и клонируете в /opt
то это неверное действие
читайте про простую установку на опенврт

**Comment by Mihrutkin** [comment]
Там 8 мб свободных остается после прошивки. Можно вас попросить ткнуть на простую инструкцию по накату opewrt

**Reply by bol-van** [reply]
этого более чем достаточно
инсталятор запрета из   /tmp перепишет около 600 кб
еще доп пакеты
суммарно 1.5-2 мб

все что качается с репы не надо класть в /opt

**Reply by Mihrutkin** [reply]
Я делаю вот по этой инструкции и после второго шага установки программ места не остается:
opkg update
opkg install iptables-mod-extra iptables-mod-nfqueue iptables-mod-filter iptables-mod-ipopt iptables-mod-conntrack-extra ipset curl ip6tables-mod-nat gzip grep git-http
mkdir /opt
cd /tmp
git clone --depth 1 https://github.com/bol-van/zapret
cd zapret/
./install_easy.sh

**Comment by Mihrutkin** [comment]
Вот так получается чистая система до начала установки по инструкции выше
![7d1e0326ab46b24e7c3625c760e5e863659024af_2_517x207](https://github.com/user-attachments/assets/e153c8d7-46d5-4241-924b-1df219eb3595)

**Reply by bol-van** [reply]
инструкция негодная. гит все сожрет.
ищите другую если мои не осилить

**Comment by Mihrutkin** [comment]
Я прочитал вашу простую установку.
Можете подсказать команду как "Копируем zapret на роутер в /tmp"

**Reply by bol-van** [reply]
в  docs/manual_setup кое что есть

**Comment by kotsmotritnastul** [comment]
я могу порекомендовать роутер ASUS RT-AX53U который мне вчера пришёл, я его успешно прошил, установил запрет и расширил ему дисковое пространство за счёт флешки.

если будешь брать другой роутер то советую перед покупкой роутера проверить его совместимость вот тут https://openwrt.org/toh/start там надо будет набрать название модели и если оно есть в таблице то роутер должен по идее прошиваться. но конкретно асус который у меня прошивается элементарно - указываешь в настройках доступ к SSH через LAN, потом заходишь туда со своим логином и паролем для вебморды роутера и устанавливаешь опенврт как написано в инструкции https://openwrt.org/toh/asus/rt-ax53u#installation_with_mtd-write

с другим роутером я советую проверить есть ли в таблице выше страничка этого роутера и если есть то внимательно прочитать инструкцию по установке до покупки роутера, а то мало ли.

а ещё перед установкой запрета можно воткнуть в роутер флешку и сделать флешку основным диском куда будет всё устанавливаться, это можно сделать по инструкции https://openwrt.org/docs/guide-user/additional-software/extroot_configuration тогда вообще не будет никаких проблем со свободным местом, но даже без него мне хватило 26 мегабайт чтобы установить и запрет и пакеты для PPPoE и L2TP.

наверняка на других роутерах всё это тоже делается, но за этот роутер я могу поручиться потому что на нём я всё это сделал и всё заработало безупречно.

![image](https://github.com/user-attachments/assets/420a2d47-e994-4cb3-a87a-68e3ea8ffa5f)

на флешке занято 53 мегабайта потому что я так понимаю это и содержимое /overlay/ и содержимое /rom/

**Comment by Evas1oN** [comment]
Если повезет выхватить на вторичке Xiaomi Mi WiFi Router 3G v1 (именно v1, у него USB 3.0). 128/256 памяти, USB 3.0, гигабитный. На 4пда есть огромная тема связанная с ним.

**Reply by kbnchk** [reply]
Поднял даже на  Mi WiFi Router 3 (не 3g) на MT7620A . Правда пришлось собирать свою версию китайского форка openwrt - x-wrt, у них на дефолтной сборке куча мусора.

**Reply by NerdzzyDev** [reply]
Поднял на Xiaomi 4Av2 , флешки нет, но ставится без проблем

**Reply by wwsilin** [reply]
> Поднял на Xiaomi 4Av2 , флешки нет, но ставится без проблем

С этим роутером главное не вляпаться в ревизию rd05, которая никак не шьется. А так да, 4Av2 не смотря на всего 16мб флеша вполне пригоден.

**Comment by electrifying** [comment]
Любой роутер с usb портом от кинетик

**Reply by TurboKirill** [reply]
Модели Keenetic 4G (KN-1210/1211/1213), Launcher (KN-1221), Carrier (KN-1721) и Extra (KN-1714) оснащены USB-разъемом для работы исключительно с совместимыми USB-модемами 3G/4G. Подключение USB-принтеров и накопителей в этих моделях не поддерживается. Исключением являются модели Hero 4G (KN-2310/2311) и 4G (KN-1212), оснащенные многофункциональным портом USB.

**Reply by lotigara** [reply]
> Модели Keenetic 4G (KN-1210/1211/1213), Launcher (KN-1221), Carrier (KN-1721) и Extra (KN-1714) оснащены USB-разъемом для работы исключительно с совместимыми USB-модемами 3G/4G. Подключение USB-принтеров и накопителей в этих моделях не поддерживается. Исключением являются модели Hero 4G (KN-2310/2311) и 4G (KN-1212), оснащенные многофункциональным портом USB.

На некоторые Keenetic-и можно OpenWrt поставить. И тех, на которые можно поставить, становится больше

**Comment by z0mb1e-kgd** [comment]
Всё гораздо проще. 16 MB флэша хватит с головой.
Ставить лучше на чистую систему (factory reset), чтобы очистить /overlay от мусора после предыдущих экспериментов.
Подключаешься через SSH к роутеру.

`opkg update`
`opkg install curl unzip`
`cd /tmp`
`curl -L -o zapret.zip https://github.com/bol-van/zapret/archive/refs/heads/master.zip`
`unzip zapret.zip`
`opkg remove unzip`
`cd zapret-master`
`./install_prereq.sh`  # здесь нужно будет выбрать файервол и стек IP (я выбрал nftables и IPv6 -> N (No))
`./install_easy.sh`  # скрипт спросит, переносить ли zapret в локацию /opt/zapret, обязательно отвечаешь Y (YES). Далее выбираешь необходимые настройки. В конце рекомендую выбрать autohostlist. Читай мануал, если что-то при работе этого скрипта непонятно.

Далее - редактируем конфиги. Надеюсь, ты умеешь работать в редакторе vi, если нет - можно поставить редактор попроще, nano: `opkg install nano`, но нужно понимать, что он займет минимум 300 KB на носителе (свободное место на носителе можно проверить командой `df -h`, значение Available напротив точки монтирования `/`).
Для работы ютюба добавляешь в файл `/opt/zapret/ipset/zapret-hosts-user.txt` хосты [отсюда](https://gist.github.com/MashinaMashina/58eed9128ade2c8ff50a84d523ae97d1) и правишь основной конфигурационный файл `/opt/zapret/config` в соответствии с [инструкцией](https://github.com/bol-van/zapret/discussions/200) автора. Если на твоем провайдере стратегия настройки `NFQWS_OPT_DESYNC="--dpi-desync=split"` в этом файле не работает или работает плохо, запускаешь последовательно:
`/etc/init.d/zapret stop`
`/opt/zapret/blockcheck.sh`  # скрипт тестирует стратегии обхода DPI, выбираешь подходящую с пометкой !!! AVAILABLE !!!
`/etc/init.d/zapret start`
В моем случае лучше всего подошла стратегия `NFQWS_OPT_DESYNC="--dpi-desync=fake,disorder2 --dpi-desync-fooling=md5sig"`.
Далее - перезагрузка, чтобы очистить /tmp от установочных файлов:
`reboot`
Ждешь загрузки роутера и проверяешь. Если не помогло, ставь другую стратегию из `blockcheck.sh` в `/opt/zapret/config`, а после этого - `/etc/init.d/zapret restart`.
Всё. Если нужна помощь - стучи в телегу `@z0mb1ekgd`

**Reply by Mihrutkin** [reply]
Спасибо тебе добрый человек, что без высокомерия просто помог с инструкцией.

**Reply by ShNURoK42** [reply]
`./install_bin.sh` эту команду можно не выполнять, при выполнении `./install_easy.sh` создается линковка, да и в этом месте она будет в `/tmp` что не имеет смысла.

**Reply by z0mb1e-kgd** [reply]
> `./install_bin.sh` эту команду можно не выполнять, при выполнении `./install_easy.sh` создается линковка, да и в этом месте она будет в `/tmp` что не имеет смысла.

@ShNURoK42 И правда, можно не выполнять. Я просто изначально предполагал, что ./install_bin.sh подготавливает только требуемые исполняемые файлы для копирования на носитель, чтобы не копировать лишнего, но раз это не так - можно этот пункт пропустить. Поправил инструкцию.

**Comment by lsahnicne** [comment]
Прошу помощи в том же вопросе, что и ОП, поскольку текущий tp-link archer c7 v4, судя по показаниям load average и тормозам веб-интерфейса, с трудом вытягивает установленные сервисы.

По хотелкам - чтобы поддерживал OpenWRT, имел не менее 4 LAN портов и продавался в РФ.

Уже рассмотрел и отмёл следующие варианты:
1. Xiaomi Router AX3200 RB01 - 3 LAN порта.
2. TP-link archer AX23 - насколько я понял, стоящий в нём процессор MediaTek MT7621DAT не значительно производительнее, чем стоящий в текущем QCA9563
3. Xiaomi Router AX3000T - 3 LAN порта

С роутеами Xiaomi печальная картина. Привлекательны ценой и производительностью, но скупятся на порты =(

**Reply by lvm04** [reply]
Mikrotik hAP ax2 - 5 портов: https://openwrt.org/toh/mikrotik/hap_ac2
Советую смотреть роутеры, которые легко прошиваются на OpenWrt c достаточным объемом RAM/Flash. Количество LAN портов можно увеличить путем подключения к роутеру самого дешевого доп. свитча на 5-8 портов.

**Reply by lsahnicne** [reply]
Глядя на объём памяти, частоту процессора и отсутствие многоядерности, по сравнению с [моим арчером](https://openwrt.org/toh/tp-link/archer_c7#hardware_highlights) этот микротик на мой неопытный взгляд выглядит скорее сайдгрейдом, а не апгрейдом.
А со свичом хорошая идея, что-то я не подумал. Спасибо за совет!

**Comment by kbnchk** [comment]
Лучший роутер из всех на которых я гонял запрет это Friendly Elec R2S. Очень рекомендую, оперативки 1гб, быстрый арм.

**Reply by lsahnicne** [reply]
1 LAN порт и отсутствие беспроводной связи? Я бы со своей обывательской точки зрения это роутером не назвал. Скорее коробка, которая ставится в разрыв в качестве дополнительной вычислительной мощности.

**Reply by kbnchk** [reply]
Роутер - это то что осуществляет l3 routing, может быть хоть виртуальной машиной (привет pfsense). Не вижу проблем сунуть в этой коробчке в лан любой wifi access point или роутер в режиме полного бриджа без ната. Я воообще думаю купить хороший л2 свич и хороший аксеспоинт. Мне это больше нравится чем страдание с каличными 128 мб огрызками у которых softirq d в полку ложится при работе какого-нибудь dnscrypt2-proxy или при полной загрузке канала. Но тут на вкус и цвет. Многие действителдьно люябят комбайны, плохо выполняющие все три функции в одном.

**Comment by TurboKirill** [comment]
Я бы Кинетики рекомендовал, но надо где не менее 128мб памяти, т.к. это с бекапом и реальная делим на 2, а по факту свободной памяти 46мб и после установки запрета занято 31мб. Из плюсов, что не надо прошивать, всё ставится и работает со стандартной КинетикОС с доустановкой Entware, файлы конфига и листы доменов можно редактировать в блокноте и просто перекидывать мышкой в роутер без всяких консолей и ssh, что может напугать новичков. Установка тоже простая, всё работает без каких-либо дополнений "под кинетик". Минус только их цена (за версии от 128мб). А из дешевых есть роутеры от Билайна на вторичке, но там уже танцы с прошивкой.

**Reply by LimpingK** [reply]
Плюсую. Только что накатил nfqws на keenetic, это волшебно! При том, что был опыт с ByeDPI на Android, но ощущение что там эксплуатировать было сложнее, чем тут.
Перед этим пробовал накатить на роутер с OpenWRT, и там упёрся в ламерские проблемы от незнания линя.
P.S. Когда платите на Кинетик - вы платите не только за железо. Вы платите за то, что даже для устройств которым по 7 лет получаете обновления прошивки с вылизыванием работы DNS и т.п. - то, что реально важно.
https://t.me/keenetic_firmware/1007

**Comment by Mihrutkin** [comment]
Так есть же routerich за 4 тыс.
А кинетик - ultra стоят 19 тыс, а нет 2.5 gbit портов.
Ну такое.

**Reply by TurboKirill** [reply, marked as answer]
есть Вива за 8 тыс., тоже 128мб

**Reply by Mihrutkin** [reply]
Правильно, купить за 8 тыс роутер с AC вместо routerich AX за 4 тыс

**Reply by lsahnicne** [reply]
Мутно как-то выглядят эти роутеричи, с дистрибуцией исключительно через Telegram.

**Reply by electrifying** [reply]
> А кинетик - ultra стоят 19 тыс, а нет 2.5 gbit портов.

Ну вообще-то есть

**Reply by Mihrutkin** [reply]
Один ван порт? Ну да ну да

**Comment by kbnchk** [comment]
Да, еще ставил zapret на несколько Asus RT-AX53U (оно же AX-1800U) полет нормальный. Да, с 256 памяти не разбежишься, но аппаратная разгрузка работает вполне нормально.

**Comment by ghost** [comment]


**Reply by Mihrutkin** [reply]
Ну куда ему до xiaomi ax3000t и routerich 3000
